metrics/memoryusage.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2015 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/zenoss/glog"
+)
+
+type ServiceInstance struct {
+	ServiceID  string
+	InstanceID int
+}
+
+type MemoryUsageStats struct {
+	StartDate  time.Time
+	EndDate    time.Time
+	HostID     string
+	ServiceID  string
+	InstanceID string
+	Last       int64
+	Max        int64
+	Average    int64
+}
+
+func convertMemoryUsage(data *PerformanceData) []MemoryUsageStats {
+	mems := make([]MemoryUsageStats, len(data.Results))
+	for i, result := range data.Results {
+		mems[i] = MemoryUsageStats{
+			StartDate: time.Unix(data.StartTimeActual, 0),
+			EndDate:   time.Unix(data.EndTimeActual, 0),
+		}
+
+		for tag, value := range result.Tags {
+			switch tag {
+			case "controlplane_host_id":
+				mems[i].HostID = value[0]
+			case "controlplane_service_id":
+				mems[i].ServiceID = value[0]
+			case "controlplane_instance_id":
+				mems[i].InstanceID = value[0]
+			}
+		}
+
+		var last, sum, max float64
+		for _, dp := range result.Datapoints {
+			// lets skip NaN values
+			if dp.Value.IsNaN {
+				continue
+			}
+
+			if last = dp.Value.Value; last > max {
+				max = last
+			}
+			sum += dp.Value.Value
+		}
+		mems[i].Last = int64(last)
+		mems[i].Max = int64(max)
+		mems[i].Average = int64(sum / float64(len(result.Datapoints)))
+	}
```

Comment:
I guess we'd want to use some proper rounding function here, but probably doesn't matter for single byte values.